CIP-XXXX/README.md

```diff
@@ -0,0 +1,230 @@
+
+Thus, the goal of the developers is to ensure that the set of UTxOs available to the wallet and the set of UTxOs the backend cardano-node knows about are synchronized enough to not cause errors when a wallet or backend operations are performed. To give a few examples of potential issues, consider the following scenarios:
+
+
+- A dApp tries to balance a transaction with UTxOs from the wallet that is not yet available in dApp backend's cardano node, causing an error response during execution units evaluation
+- A transaction is passed for signing, but the wallet does not yet know about the UTxOs it spends, and thus refuses to sign it
+- A transaction is sent to the network via the dApp backend (bypassing CIP-30 submit method) and is confirmed there, but the wallet still does not know about its consumed inputs, and thus returns outdated data.
+
+### Principles of this CIP
+
+In order to build a well-thought standard, we must first establish our principles and then specify the APIs according to these principles, users' needs and the scope of the standard.
+
+#### Layered control over transaction building
+
+Different users require different levels of control over transaction building process, for reasons such as optimising fees, ensuring proper ordering of transaction inputs, or backwards compatibility.
+
+There are three conceptual levels of control:
+
+- **CL1.** transaction constraints level - the user only cares about particular requirements a transaction must satisfy in order to be valid in the context of the app, such as sending a certain amount to an address, consuming a UTxO, or submitting a stake delegation.
+- **CL2.** transaction contents level - the user cares about particular details of transaction structure, such as number and contents of change outputs, or ordering of inputs.
+- **CL3.** CBOR level - the user cares about cbor layout of the transaction, in particular, about key ordering in maps and definite/indefinite length serialization format of CBOR arrays
+
+The goal of this standard is to satisfy the needs of everyone, however, we don't want to force the users who want to work on a high level to deal with lower levels of abstraction. This would be a major advantage over CIP-30, that delegates transaction building to the user completely.
+
+However, we also don't aim at providing a more *restrictive* interface. Providing a single high-level interface would not be enough for users with extra requirements: such a standard would be a regression from CIP-30.
+
+#### Statelessness
+
+All APIs must be stateless for ease of debugging: assuming on-chain state is constant, all API endpoints must return the same responses regardless of any previous API calls, with the only exception being rate limiting errors.
+
+## Specification
+
+### Query layer
```

Comment:
Should wallets be treated as a all-knowing query layer for Cardano?
Do wallet providers want to support this? - could it be expensive for them?

Reply:
:100:, maybe part of the query layer that allows querying arbitrary (not just user's wallet-related) data such as any utxo or transaction could be a separate extension/cip. Not sure how to define the boundary here.

Reply:
> Should wallets be treated as a all-knowing query layer for Cardano?
> Do wallet providers want to support this? - could it be expensive for them?

@Ryun1 I addressed these concerns in my new CPS: https://github.com/cardano-foundation/CIPs/pull/625

Reply:
> Should wallets be treated as a all-knowing query layer for Cardano?

Standardizing the query layers would allow wallets to be free from the burden of running their own infrastructure. Thus it would not be a problem to force the wallets to provide query layers - they will be just passing data from JS to query layer provider's HTTP interface, which does not require any infra.